### ISSUE TITLE
Invalid State: Connection already closed (again)

### ISSUE THREAD

**Issue by steel-yourself** [mongo-dart/mongo_dart#305]
While trying to connect to an old Mongo 3.2 server (not local) we keep encountering this error when trying to open the database:

`MongoDB ConnectionException: Invalid state: Connection already closed.`

The connection string is in the format **mongodb://** since this is a legacy system.  We have tried using both **Db(uriString)** and **Db.create(uriString)** to open the database, but neither works.

This is our code:

```
var mongoOLD = await Db.create('mongodb://<IP ADDRESS>:27017/<DATABASE>');
await mongoOLD.open(writeConcern: WriteConcern.acknowledged);
```

According to the documentation, it seems that this package _should_ work with < Mongo 3.6 databases.  Do later versions of this package no longer support those?  Or is there something else required to connect to older versions of MongoDB databases?

(I have looked at issue [#284](https://github.com/mongo-dart/mongo_dart/issues/284), but the person never made a follow-up reply.)

**Comment by giorgiofran** [comment]
Theoretically it should work. I will put some restrictions on the mongoDb version level only starting from version 1.0 of the driver (but it still has to come).
I will tell you the truth, as I thought that nobody was still using version 3.2, recently I did my tests only starting from release 3.6.
Try to set a finer logging level and see if some interesting message is logged.

**Comment by steel-yourself** [comment]
Thanks for the logging level tip, giorgiofran.  Upon reviewing the log messages, here's what I'm seeing:

`AssertionException handling request, closing client connection: 34348 cannot translate opcode 2013`

So, it would seem that older versions of Mongo < 3.6 actually _don't_ work, since **Opcode 2013 (OPMSG)** wasn't introduced until Mongo 3.6?  Is there a different flag or anything that can be set when connecting to coerce  **.open()** into using a different messaging format?

**Comment by giorgiofran** [comment]
Version 0.8.1 is published with the fix for rel 3.2. Please give it a try and let me know if eveything is ok.

**Comment by giorgiofran** [comment]
you are welcome
